ampd/src/evm/finalizer.rs

```diff
@@ -161,4 +171,36 @@ mod tests {
                 .unwrap()
         );
     }
+
+    #[test]
+    async fn pick_should_work() {
+        let mut rpc_client = MockEthereumClient::new();
+        let mut block = Block::<Hash>::default();
+        let block_number: U64 = 10.into();
+        block.number = Some(block_number.clone());
+        let pow_confirmation_height = 6;
+
+        rpc_client
+            .expect_finalized_block()
+            .returning(move || Ok(block.clone()));
+        rpc_client
+            .expect_block_number()
+            .returning(move || Ok(block_number));
+
+        let finalizer = pick(
+            &FinalizerType::Ethereum,
+            &rpc_client,
+            pow_confirmation_height,
+        );
+        assert_eq!(
+            finalizer.latest_finalized_block_height().await.unwrap(),
+            block_number
+        );
+
+        let finalizer = pick(&FinalizerType::PoW, &rpc_client, pow_confirmation_height);
+        assert_eq!(
+            finalizer.latest_finalized_block_height().await.unwrap(),
+            block_number - U64::from(pow_confirmation_height - 1)
+        );
```

Comment:
I'd separate these into different tests to prevent this test from growing in the future, i.e. `pick_should_work_for_ethereum_finalizer` and `pick_should_work_for_pow_finalizer`.